### PR TITLE
feat(forms): always show the helper text under controls

### PR DESCRIPTION
**Pull request by dtsanevmw**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Helper text is not shown in error / success state.

Issue Number: CDE-700

## What is the new behavior?

Helper text is now always visible.

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

## Other information

BREAKING CHANGE: helper text under controls will be always visible and `clr-subtext` should be combined with `error` or `success` when used without Clarity components. Example: `<div class="clr-subtext error">Error message</div>`

### CODE REVIEW

File: projects/angular/src/forms/common/abstract-container.ts [hunk 1]
```diff
@@ -54,27 +54,32 @@ export abstract class ClrAbstractContainer implements DynamicWrapper, OnDestroy,
...
+   * Helper control is a bit different than the others, it must be always visible:
+   *   -  Labels and instructions must always accompany forms and are persistent.
+   *   -  The recommendation here is to always have helper text or anything instructions visible.
+   *   -  The expectation is to have error text + helper text in the errored state. this way all users will have the helper text information always available.
    */
   get showHelper(): boolean {
     // without existence of helper component there is no need of additional checks.
     if (!!this.controlHelperComponent === false) {
       return false;
     }
```
Comment (LynxDelli) [inline projects/angular/src/forms/common/abstract-container.ts:67]: You can also simply return this.controlHelperComponent without the if :)

**Comment by kevinbuhmann** [comment]
I think this should be marked as a breaking change. It is visually breaking and may require some helper/error texts to be rewritten so that both make sense together abandoned aren't redundant.

**Comment by dtsanevmw** [comment]
> I think this should be marked as a breaking change. It is visually breaking and may require some helper/error texts to be rewritten so that both make sense together abandoned aren't redundant.

Yeah, I reverted it to draft due to issues people can have if they are not using the components and just the css classes.

**Review by valentin-mladenov — COMMENTED** [review]
LGTM, but To meet some visual regressions.

Changes in [_containers.clarityscss#L145](https://github.com/vmware-clarity/ng-clarity/blob/dtsanevmw/cde-700/projects/angular/src/forms/styles/_containers.clarity.scss#L145) must be made. Enhance the rule by adding 2 more. 

```
  .clr-form:not(.clr-form-compact) {
    .clr-error,
    .clr-success {
      .clr-checkbox-wrapper,
      .clr-file-wrapper,
      .clr-radio-wrapper {
        & > .clr-subtext-wrapper,
        & > .clr-validate-icon,
        & + .clr-subtext-wrapper,
        & + .clr-validate-icon {
```

Update the branch from next first, there are some changes in that file on those lines

**Review by valentin-mladenov — APPROVED** [review]
LGTM

**Pull request merged**